android/libraries/rib-base/src/main/java/com/badoo/ribs/core/routing/configuration/feature/BackStackFeature.kt

```diff
@@ -29,23 +46,49 @@ private fun <C : Parcelable> TimeCapsule<BackStackFeatureState<C>>.initialState(
  * @see BackStackFeature.ActorImpl for logic deciding whether an operation should be carried out
  * @see BackStackFeature.ReducerImpl for the implementation of applying state changes
  */
-internal class BackStackFeature<C : Parcelable>(
-    initialConfiguration: C,
+class BackStackFeature<C : Parcelable>(
+    private val initialConfiguration: C,
     timeCapsule: TimeCapsule<BackStackFeatureState<C>>
-) : ActorReducerFeature<Operation<C>, Effect<C>, BackStackFeatureState<C>, Nothing>(
-    initialState = timeCapsule.initialState(),
-    bootstrapper = BootstrapperImpl(
-        timeCapsule.initialState(),
-        initialConfiguration
-    ),
-    actor = ActorImpl<C>(),
-    reducer = ReducerImpl<C>()
-) {
+) : Consumer<Operation<C>>, RoutingSource<C>, BackPressHandler {
+
+    val feature = ActorReducerFeature<Operation<C>, Effect<C>, BackStackFeatureState<C>, Nothing>(
+        initialState = timeCapsule.initialState(),
+        bootstrapper = BootstrapperImpl(
+            timeCapsule.initialState(),
+            initialConfiguration
+        ),
+        actor = ActorImpl(),
+        reducer = ReducerImpl()
+    )
+
+    val state: BackStackFeatureState<C>
+        get() = feature.state
+
+    val activeConfiguration: ObservableSource<C>
+        get() = Observable.wrap(feature)
+            .mapNotNull {
+                it.backStack
+                    // FIXME set active in operation
+    //                .findLast { it.activation == RoutingHistoryElement.Activation.ACTIVE }
+                    .last()
+                    ?.routing
+                    ?.configuration
+            }
+            .startWith(initialConfiguration)
+
+    constructor(
+        initialConfiguration: C, // FIXME this should be RoutingHistoryElement<C>
```

Comment:
It will be another breaking changes?

Reply:
won't do now, changed wording to consider -- can be done with an additional constructor without breaking changes too